CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/icon/DefaultCustomIconProviderNGTest.java

```diff
@@ -0,0 +1,196 @@
+ *
+ * @author Delphinus8821
+ */
+public class DefaultCustomIconProviderNGTest {
+
+    final static String TEST_ICON_NAME = "TestIcon2";
+
+    public DefaultCustomIconProviderNGTest() {
+    }
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+    }
+
+    /**
+     * Runner for tests
+     */
+    @Test(expectedExceptions = NullPointerException.class)
+    public void runTests() {
+        try ( MockedStatic<DefaultCustomIconProvider> defaultCustomIconProviderMock = Mockito.mockStatic(DefaultCustomIconProvider.class)) {
+            try ( MockedStatic<IconManager> iconManagerMock = Mockito.mockStatic(IconManager.class)) {
+                // Get a test directory location for the getIconDirectory call
+                URL exampleIcon = DefaultCustomIconProviderNGTest.class.getResource("resources/");
+                File testFile = new File(exampleIcon.toURI());
+                defaultCustomIconProviderMock.when(() -> DefaultCustomIconProvider.getIconDirectory()).thenReturn(testFile);
+
+                // Create a test icon to be removed by testRemoveIcon
+                final ConstellationColor ICON_COLOR = ConstellationColor.BLUEBERRY;
+                final ConstellationIcon ICON_BACKGROUND = DefaultIconProvider.FLAT_SQUARE;
+                final ConstellationIcon ICON_SYMBOL = AnalyticIconProvider.STAR;
+
+                ConstellationIcon icon = new ConstellationIcon.Builder(TEST_ICON_NAME,
+                        new ImageIconData((BufferedImage) ImageUtilities.mergeImages(
+                                ICON_BACKGROUND.buildBufferedImage(16, ICON_COLOR.getJavaColor()),
+                                ICON_SYMBOL.buildBufferedImage(16), 0, 0)))
+                        .build();
+                icon.setEditable(true);
+                DefaultCustomIconProvider instance = new DefaultCustomIconProvider();
+                // Add a test icon to be removed later
+                instance.addIcon(icon);
+
+                // Run testAddIcon
+                testAddIcon(iconManagerMock);
+
+                //Run testAddIconFileDoesExist
+                testAddIconFileDoesExist(icon);
+
+                // Run testRemoveIcon
+                testRemoveIcon(iconManagerMock, icon);
+
+                // Run testRemoveIconDoesNotExist
+                testRemoveIconDoesNotExist();
+
+                // Run testLoadIcons
+                testLoadIcons(testFile);
+
+                // Verify defaultCustomIconProvider.getIconDirectory was called the correct number of times 
+                defaultCustomIconProviderMock.verify(() -> DefaultCustomIconProvider.getIconDirectory(), times(6));
+
+            } catch (final URISyntaxException ex) {
```

Comment:
I would probably throw this exception rather than catching it if all that is being done is printing the stack trace (you probably want the test to fail in this case anyway)